src/eva/data/mon_data_space.py

```diff
@@ -200,87 +196,279 @@ def subset_coordinate(self, ds, coordinate, requested_subset):
+                                                                          dims['ydef'])
+                        mylist.append(arr)
+                    tarr = np.dstack(mylist)
+
+                rtn_array = np.append(rtn_array, [tarr], axis=0)
 
         f.close()
+        return rtn_array, cycle_tm
+
+    # ----------------------------------------------------------------------------------------------
 
-        return count, penalty, omgnbc_sum, total_sum, omgbc_sum, \
-            omgnbc_sum2, total_sum2, omgbc_sum2, cycle_tm
+    def cycle_to_np_array(self, dims, ndims_used, cycle_tm):
+
+        # build numpy array with requested dimensions
+        cycle_arr = None
+
+        if ndims_used == 3:
+            cycle_arr = np.reshape([[cycle_tm] * dims['xdef'] * dims['ydef'] * dims['zdef']],
+                                   (dims['xdef'], dims['ydef'], dims['zdef']))
+        elif ndims_used == 2:
+            cycle_arr = np.reshape([[cycle_tm] * dims['xdef']] * dims['ydef'],
+                                   (dims['xdef'], dims['ydef']))
+        elif ndims_used == 1:
+            cycle_arr = np.reshape([[cycle_tm] * dims['xdef']], (dims['xdef']))
+        else:
+            self.logger.abort(f'ndims_used must be in range of 1-3, value is {ndims_used}')
+        return cycle_arr
 
     # ----------------------------------------------------------------------------------------------
+
+    def get_dim_ranges(self, coords, dims, channo):
+        x_range = None
+        y_range = None
+        z_range = None
+
+        # - Return None for a coordinate that has value 0 or 1.
+        # - "Channel" can be either the x or y coordinate and can be
+        #      numbered non-consecutively, which has been captured in channo.
+        # - The z coordinate is never used for channel.
+
+        if dims['xdef'] > 1:
+
+            if coords['xdef'] == 'Channel':
+                x_range = channo
+            else:
+                x_range = np.arange(1, dims['xdef']+1)
+
+        if dims['ydef'] > 1:
+            if coords['ydef'] == 'Channel':
+                y_range = channo
+            else:
+                y_range = np.arange(1, dims['ydef']+1)
+
+        if dims['zdef'] > 1:
+            z_range = np.arange(1, dims['zdef']+1)
+
+        return x_range, y_range, z_range
+
+    # ----------------------------------------------------------------------------------------------
+
+    def get_ndims_used(self, dims):
+
+        # Ignore dims with values of 0 or 1
+        ndims = len(dims)
+        for x in range(ndims):
+            if list(dims.values())[x] <= 1:
+                ndims -= 1
+
+        return ndims
+
+    # ----------------------------------------------------------------------------------------------
+
+    def load_dset(self, vars, nvars, coords, darr, dims, ndims_used,
+                  x_range, y_range, z_range, cyc_darr):
+
+        # create dataset from file components
+        rtn_ds = None
+
+        if ndims_used == 1:
```

Comment:
Similar comment to above, I think these may be able to be consolidated a bit, but 4:15pm Friday brain doesn't have the exact solution

Reply:
I've got an idea here.  I think if I create a dictionary from the variable dimensions and coordinates I can then create the dataset via `xr.Dataset.from_dict(d)`.  That would put the if `num_dims` logic inside the for loop.  Giving it a shot now.

Reply:
See if you like this solution.

Reply:
looks good!